lib/index.js

```diff
@@ -686,14 +701,27 @@ function _openDatabase(options, callback) {
       const username = config.username;
       const password = config.password;
       let opts = config.authentication;
+      const {MongoClient, Server} = mongo;
       if(_usesRoles(results.serverInfo)) {
         // authenticate against shared db
-        opts = bedrock.util.extend({}, opts, {authdb: config.name});
-        return db.authenticate(username, password, opts, callback);
+        opts = bedrock.util.extend({}, opts, {authSource: config.name});
```

Comment:
This would do the same, right? We should phase out `bedrock.util.extend` where new language options are simpler.

```suggestion
        opts = {...opts, authSource: config.name};
```

Reply:
This was addressed here: https://github.com/digitalbazaar/bedrock-mongodb/pull/33/commits/1c76605ece2b2c3a295ccfe074866b716281e0de